src/st/class-wpml-pb-integration.php

```diff
@@ -195,18 +195,60 @@ public function do_shutdown_action() {
 
 	public function new_translation( $translated_string_id ) {
 		if ( ! $this->is_registering_string ) {
-			foreach ( $this->strategies as $strategy ) {
+			$this->with_strategies( function( $strategy ) use ( $translated_string_id ) {
 				$this->factory->get_string_translations( $strategy )->new_translation( $translated_string_id );
-			}
+			} );
 			$this->new_translations_recieved = true;
 		}
 	}
 
+	/**
+	 * @param callable $callable
+	 * 
+	 * @return mixed
+	 */
+	private function with_strategies( callable $callable ) {
+		foreach ( $this->strategies as $strategy ) {
+			return $callable( $strategy );
+		}
+	}
+
+	/**
+	 * When a Page Builder content has only a "LINK" string, it's won't be part
+	 * of the translation job as it's automatically converted.
+	 * We need to add the package to the update list (by strategies).
+	 *
+	 * @param int $new_post_id
+	 * @param int $original_doc_id
+	 */
+	public function process_pb_content_with_hidden_strings_only( $new_post_id, $original_doc_id ) {
+		if (
+			! did_action( 'wpml_add_string_translation' )
+			&& apply_filters( 'wpml_pb_is_page_builder_page', false, get_post( $new_post_id ) )
+		) {
+			$targetLang = $this->sitepress->get_language_for_element( $new_post_id, 'post_' . get_post_type( $new_post_id ) );
+
+			$addPackageToUpdateList = $this->with_strategies( function( IWPML_PB_Strategy $strategy ) use ( $targetLang ) {
+				return function( WPML_Package $package ) use ( $strategy, $targetLang ) {
+					$this->factory
+						->get_string_translations( $strategy )
+						->add_package_to_update_list( $package, $targetLang );
+				};
+			} );
+
+			wpml_collect( apply_filters( 'wpml_st_get_post_string_packages', [], $original_doc_id ) )
+				->each( $addPackageToUpdateList )
+				->first( function() {
```

Comment:
@strategio I think you're mis-using the `first` function here. 

Maybe you could do:

```php
if ( wpml_collect( apply_filters( 'wpml_st_get_post_string_packages', [], $original_doc_id ) )
				->each( $addPackageToUpdateList )
				->isNotEmpty() ) {
    $this->new_translations_recieved = true;
}
```

Reply:
Thanks @brucepearson, even better:

```php
$this->new_translations_recieved = wpml_collect( apply_filters( 'wpml_st_get_post_string_packages', [], $original_doc_id ) )
				->each( $addPackageToUpdateList )
				->isNotEmpty();
```